### ISSUE TITLE
Linter should check for “.” in p and li

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/respec#1572]
Got a code review where I had forgotten full stops in a couple of places. That wastes reviewers time.

**Comment by OrionStar25** [comment]
```
let var = doc.querySelector("p");
expect(var).toBeTruthy();
expect(var.textContent).toMatch($("p").text() + /./);

Is this what needs to be done essentially?
```

**Comment by marcoscaceres** [comment]
something like that, but it would be a bit more involved: basically, I would like to asynchronously (via `requestIdleCallback()`, if available) check each `p.textContent.trim().endsWith(".")`.

**Comment by OrionStar25** [comment]
Alright, excellent! I'd like to work on this :100:

**Comment by marcoscaceres** [comment]
Ok, please take a look at:
https://github.com/w3c/respec/blob/develop/src/core/linter.js

And the linterRules directory: 
https://github.com/w3c/respec/blob/develop/src/core/

But let’s finish the biblio PR before starting on this one.

**Comment by shikhar-scs** [comment]
@marcoscaceres can I please take up this issue if no one is working on it ?

**Comment by marcoscaceres** [comment]
@shikhar-scs sure, but coordinate with @OrionStar25, as she also wanted to help. 

If you do start work on it, you might want to follow this one:
https://github.com/w3c/respec/blob/develop/src/core/linter-rules/no-http-props.js

Also, we should start with just the `p` elements, as I think they are less difficult than the `li`. The `li` checks would  be a separate linter rule - as some spec writers prefer to exclude "." from bulleted lists in certain situations). 

We should check this list of specs:
https://github.com/w3c/respec/blob/develop/tests/headless.js#L4

To see if there are odd edge cases where missing a "." might be ok.

**Comment by shikhar-scs** [comment]
Awesome ! I'll get started with this :+1:

**Comment by xfq** [comment]
Should we close this given that #1589 has been merged?

By the way, I think we should only check `.:!?` in documents with `lang=en`, because not all languages/scripts use the Latin punctuations, and we do have ReSpec documents in these languages. I can file another issue for this if needed.

Some examples of full stops:

* "。" in Chinese (`zh-hans` and `zh-hant`) and Japanese (`ja`)
* "।" in Hindi (`hi`)
* "۔" in Urdu (`ur`)

**Comment by marcoscaceres** [comment]
True, let's leave this open then until we deal with the l10n issues.

**Comment by marcoscaceres** [comment]
Closing this linter rule request for checking sentence-ending punctuation. After 8 years of discussion, the consensus was that this rule would generate too many false positives on code blocks, sub-lists, and technical prose to be practical.
